code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -1071,3 +1071,23 @@
 		connected_ai.connected_robots |= src
 		notify_ai(ROBOT_NOTIFICATION_NEW_UNIT)
 		sync()
+
+//For picking up small animals
+/mob/living/silicon/robot/MouseDrop(atom/over_object)

```

Comment:
oops, i forgot to delete that comment. I copied it over from holder.dm just to get the template for the function, then deleted everything and rewrote it

The function is fully functional, i've spent almost two hours testing it very thoroughly